client/ayon_core/pipeline/traits/lifecycle.py

```diff
@@ -0,0 +1,77 @@
+"""Lifecycle traits."""
+from dataclasses import dataclass
+from typing import ClassVar
+
+from .trait import TraitBase, TraitValidationError
+
+
+@dataclass
+class Transient(TraitBase):
+    """Transient trait model.
+
+    Transient trait marks representation as transient. Such representations
+    are not persisted in the system.
+
+    Attributes:
+        name (str): Trait name.
+        description (str): Trait description.
+        id (str): id should be namespaced trait name with version
+    """
+
+    name: ClassVar[str] = "Transient"
+    description: ClassVar[str] = "Transient Trait Model"
+    id: ClassVar[str] = "ayon.lifecycle.Transient.v1"
```

Comment:
If a new version of this trait would get created, does that mean we'd need a different class? Or how would we actually version traits in practice?

I assume somehow we'd need to keep the old trait class implementation around?

Reply:
Issue is that versioning make sense in OTIO schemas or the original OAIO where we deal with json schema files, right? This way we will need to add version even to the class name, right?

Reply:
Whenever you want to de-serialize data that are using older version of trait, `upgrade()` method on newer trait definition is called to reconstruct new version (downgrading isn't possible).  That means when you encounter trait like `ayon.2d.Image.v1` in __data__ and your runtime **Image** trait is v2, some method will call `upgrade()` method on newer trait version if implemented (for example in cases where newer version knows how to provide missing data or do conversion, etc.)

Reply:
For reference - `upgrade()` is called with old trait as an argument on `.from_dict()` calls and when you get traits by id and you omit version specifier (and old trait different than the runtime version is found)